liblouis/metadata.c

```diff
@@ -199,15 +199,24 @@ cmpKeys(Feature *f1, Feature *f2) {
 	return strcasecmp(f1->key, f2->key);
 }
 
+
+static int
+cmpFeatures(Feature *f1, Feature *f2) {
+	int r = strcasecmp(f1->key, f2->key);
+	if (r == 0)
+		r = strcasecmp(f1->val, f2->val);
+	return r;
+}
+
 /**
  * Compute the match quotient of the features in a query against the features in a table's
  * metadata.
  *
- * The features are assumed to be sorted and to have no duplicate
- * keys. The query's features must be of type FeatureWithImportance.
- * How a feature contributes to the match quotient depends on its
- * importance, on whether the feature is undefined, defined with the
- * same value (positive match), or defined with a different value
+ * The features are assumed to be sorted. The query's features must be
+ * of type FeatureWithImportance and are assumed to have no duplicate
```

Comment:
Don't you allow duplicate keys now? so the sentence "and are assumed to have no duplicate keys" would not be correct.

Reply:
I only allow duplicate keys in the table metadata, not in the query.

Reply:
OK